pkg/util/provider/machinecontroller/machine.go

```diff
@@ -223,10 +235,13 @@ func (c *controller) deleteNodeToMachine(obj interface{}) {
 
 	machine, err := c.getMachineFromNode(key)
 	if err != nil {
-		klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
-		return
-	} else if machine == nil {
-		return
+		if err == multipleMachineMatchError {
+			klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
+			return
+		} else if err == noMachineMatchError {
+			klog.Errorf("Couldn't fetch machine %s, Error: %s", key, err)
+			return
+		}
```

Comment:
Does this change now make sense given that the handling for the error cases is now pretty much the same?

Reply:
@amshuman-kr  review addressed in b6a183c..brought back the discussed section of code to the old state.. please recommend code in case you feel any changes needed.